pulp_2to3_migration/app/models/repository.py

```diff
@@ -43,7 +43,7 @@ class Pulp2Repository(BaseModel):
     not_in_plan = models.BooleanField(default=False)
     type = models.CharField(max_length=25)
 
-    pulp3_repository_version = models.OneToOneField(RepositoryVersion,
+    pulp3_repository_version = models.ForeignKey(RepositoryVersion,
```

Comment:
1. run MP
```{
    "plan": {
        "plugins": [
            {
                "repositories": [
                    {
                        "name": "foo2",
                        "pulp2_importer_repository_id": "test-iso",
                        "repository_versions": [
                            {
                                "pulp2_distributor_repository_ids": [
                                    "file2"
                                ],
                                "pulp2_repository_id": "test-iso"
                            }
                        ]
                    }
                ],
                "type": "iso"
            }
        ]
    },
    "pulp_created": "2020-01-30T13:14:31.349816Z",
    "pulp_href": "/pulp/api/v3/migration-plans/a8b21a69-e3e9-4199-b858-7e737b2eb582/"
}

```
2. delete test-iso, re-create test-iso. add there back same content as it had before it was deleted
3. run MP
4. observe traceback:
```
$ http GET :24817/pulp/api/v3/tasks/54fe63b2-e76a-4229-8a88-c55f52761294/
HTTP/1.1 200 OK
Allow: GET, PATCH, DELETE, HEAD, OPTIONS
Connection: close
Content-Length: 4694
Content-Type: application/json
Date: Thu, 30 Jan 2020 13:03:05 GMT
Server: gunicorn/20.0.4
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "created_resources": [],
    "error": {
        "description": "duplicate key value violates unique constraint \"pulp_2to3_migration_pulp2reposi_pulp3_repository_version_id_key\"\nDETAIL:  Key (pulp3_repository_version_id)=(cc7846c2-8564-4df4-a5b3-ca8fcca2858a) already exists.\n",
        "traceback": "  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/rq/worker.py\", line 875, in perform_job\n    rv = job.perform()\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/rq/job.py\", line 648, in perform\n    self._result = self._execute()\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/rq/job.py\", line 654, in _execute\n    return self.func(*self.args, **self.kwargs)\n  File \"/home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/app/tasks/migrate.py\", line 115, in migrate_from_pulp2\n    loop.run_until_complete(create_repo_versions(plan))\n  File \"/usr/lib64/python3.6/asyncio/base_events.py\", line 484, in run_until_complete\n    return future.result()\n  File \"/home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/app/migration.py\", line 312, in create_repo_versions\n    create_repo_version(plugin.migrator, repo_name, pulp2_repo, pulp3_remote)\n  File \"/home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/app/migration.py\", line 273, in create_repo_version\n    pulp2_repo.save()\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/base.py\", line 741, in save\n    force_update=force_update, update_fields=update_fields)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/base.py\", line 779, in save_base\n    force_update, using, update_fields,\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/base.py\", line 851, in _save_table\n    forced_update)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/base.py\", line 900, in _do_update\n    return filtered._update(values) > 0\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/query.py\", line 760, in _update\n    return query.get_compiler(self.db).execute_sql(CURSOR)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/sql/compiler.py\", line 1462, in execute_sql\n    cursor = super().execute_sql(result_type)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/sql/compiler.py\", line 1133, in execute_sql\n    cursor.execute(sql, params)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py\", line 67, in execute\n    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py\", line 76, in _execute_with_wrappers\n    return executor(sql, params, many, context)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py\", line 84, in _execute\n    return self.cursor.execute(sql, params)\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/utils.py\", line 89, in __exit__\n    raise dj_exc_value.with_traceback(traceback) from exc_value\n  File \"/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py\", line 84, in _execute\n    return self.cursor.execute(sql, params)\n"
    },
```

RV is already assigned to the old pulp2-repo that is not in plan.


Another scenario i tested:

1. run MP
2. delete test-iso( that has 3 isos), re-create test-iso and add there 1 iso
3. run MP
4. task ok.

Reply:
one more scenario:
   1.run MP
   2. delete test-iso, re-create test-iso( leave it empty)
   3. run MP